### ISSUE TITLE
Provide an option to generate fewer (larger) pages.

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/dartdoc#1272]
We used to document members in the class that declared them, but now each member gets its own class. 

In a large API, this means thousands of additional files, which can be problematic if you do any processing on the output. For example, adding the Dart API docs to [angular.io/dart](https://angular.io/dart) increased the build time by many minutes, because almost 3000 files were added to the build. (The entire site has 3858 .jade files, of which 2845 are in the [AngularDart API doc](https://angular.io/docs/dart/latest/api/) directory.) If we had just *-class and *-library files, plus the index file, the total # of files for the Dart API docs would be around 250.

I'd like the option to keep all the documentation in the owning class.

**Comment by devoncarew** [comment]
This would be a big change - if we do this it would be safer to generate all docs w/ fewer pages, instead of having an option for it. It would make for simpler code for handling things like cross references.

**Comment by devoncarew** [comment]
When I've thought about this in the past, I've generally come to the conclusion that libraries get their own output file, and classes get their own output file, with some sort of class summarization on the library page.

**Comment by Levi-Lesches** [comment]
I'm not on the Flutter team, but in #1983 they note that Flutter would _greatly_ benefit from this, and by extension, anyone who wants to include the Flutter docs in their private documentation. I've tried a couple of times to add Flutter docs and my computer just couldn't handle it.

**Comment by sigurdm** [comment]
For reference go-doc is doing a single long html page for all the package, see for example: https://pkg.go.dev/archive/zip
I find it quite nifty to navigate.

**Comment by sigurdm** [comment]
I agree with @devoncarew - if we do this, it should probably be to only option for output. We don't want to maintain two separate outputs
